inline.h

```diff
@@ -380,6 +398,400 @@ Perl_POPMARK(pTHX)
     return *PL_markstack_ptr--;
 }
 
+/*
+=for apidoc_section $rpp
+
+=for apidoc rpp_extend
+Ensures that there is space on the stack to push C<n> items, extending it
+if necessary.
+
+=cut
+*/
+
+PERL_STATIC_INLINE void
+Perl_rpp_extend(pTHX_ SSize_t n)
+{
+    PERL_ARGS_ASSERT_RPP_EXTEND;
+
+    EXTEND_HWM_SET(PL_stack_sp, n);
+#ifndef STRESS_REALLOC
+    if (UNLIKELY(_EXTEND_NEEDS_GROW(PL_stack_sp, n)))
+#endif
+    {
+        (void)stack_grow(PL_stack_sp, PL_stack_sp, n);
```

Comment:
This is wrong, and the prototype of `PERL_STATIC_INLINE void Perl_rpp_extend(pTHX_ SSize_t n)` wrong. The retval of `stack_grow()` is being ignored, and all `pp_*()` ops that use this static inline have to re-fetch the ptr inside `PL_stack_sp`, either from `my_perl` or the OS symbol tables (no thds perl) if this grow branch executes, instead of using the new `SV**` ptr delivered in a register from `stack_grow()`.